fastly/resource_fastly_service_authorization.go

```diff
@@ -0,0 +1,115 @@
+package fastly
+
+import (
+	"context"
+
+	gofastly "github.com/fastly/go-fastly/v6/fastly"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceServiceAuthorization() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceServiceAuthorizationCreate,
+		ReadContext:   resourceServiceAuthorizationRead,
+		UpdateContext: resourceServiceAuthorizationUpdate,
+		DeleteContext: resourceServiceAuthorizationDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The ID of this service authorization.",
+			},
+			"permission": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				Default:          "full",
```

Comment:
I'm actually surprised the API _defaults_ to "full". 

I feel it would be safer to default to "read_only" and have users explicitly opt into more permissive values. 

> **NOTE**: I've reached out to the relevant API team about this as I'm interested to know if they agree that we should default to "read_only" (and if so, then we should update the API endpoint as well to reflect that).

Reply:
I agree. `read_only` would absolutely be a better default.

Alternatively, there could be *no default*, and explicitly require users to set the permissions they desire. Better to be explicit in my opinion.

Reply:
Agreed. Let's avoid the default and users have to be explicit about what they want. 👍🏻